packages/api-reference/vite.component.config.ts

```diff
@@ -0,0 +1,72 @@
+import vue from '@vitejs/plugin-vue'
+import { webpackStats } from 'rollup-plugin-webpack-stats'
+import banner from 'vite-plugin-banner'
+import cssInjectedByJsPlugin from 'vite-plugin-css-injected-by-js'
+import { defineConfig } from 'vitest/config'
+
+import licenseBannerTemplate from './license-banner-template.txt'
+import { name, version } from './package.json'
+
+function replaceVariables(template: string, variables: Record<string, string>) {
+  return Object.entries(variables).reduce((content, [key, value]) => {
+    return content.replace(new RegExp(`\\{\\{ ${key} \\}\\}`, 'g'), value)
+  }, template)
+}
+
+export default defineConfig({
+  define: {
+    'process.env.NODE_ENV': '"production"',
+  },
+  plugins: [
+    // Important: Tells vite to treat all vue sub-components as custom elements
+    vue({ customElement: true }),
```

Comment:
This part is the important, this is why we need a separate vite config, everything else is just copied from `vite.standalone.config.js`.

See https://github.com/vitejs/vite-plugin-vue/tree/main/packages/plugin-vue#using-vue-sfcs-as-custom-elements

Reply:
Should we import the other config and just overwrite the plugins attribute (and whatever is different)? Or is that confusing?

Reply:
Yeah that might be good if I can get that working